sdk/eas/src/eas.ts

```diff
@@ -1,98 +1,482 @@
 
     try {
-      // Check if the provider is available
-      await provider.getNetwork();
+      // Deploy Schema Registry first
+      const schemaRegistryResult = await this.portalClient.request(GraphQLOperations.mutations.deploySchemaRegistry, {
+        from: deployerAddress,
+        constructorArguments: {
+          forwarder: defaultForwarder,
+        },
+        gasLimit: defaultGasLimit,
+      });
+
+      const schemaRegistryResponse = schemaRegistryResult as {
+        DeployContractEASSchemaRegistry?: { transactionHash?: string };
+      };
+
+      if (!schemaRegistryResponse.DeployContractEASSchemaRegistry?.transactionHash) {
+        throw new Error("Schema Registry deployment failed - no transaction hash returned");
+      }
+
+      const schemaRegistryTxHash = schemaRegistryResponse.DeployContractEASSchemaRegistry.transactionHash;
+
+      // Wait for Schema Registry deployment and get contract address
+      let schemaRegistryAddress: Address | null = null;
+      let attempts = 0;
+      const maxAttempts = 30;
+
+      while (!schemaRegistryAddress && attempts < maxAttempts) {
+        await new Promise((resolve) => setTimeout(resolve, 1000));
+        attempts++;
+
+        try {
+          const contractsResult = await this.portalClient.request(
+            "query GetSchemaRegistryContracts { getContractsEasSchemaRegistry { records { address } } }",
+          );
+
+          const contractsResponse = contractsResult as {
+            getContractsEasSchemaRegistry?: { records: Array<{ address: string }> };
+          };
+
+          const contracts = contractsResponse.getContractsEasSchemaRegistry?.records || [];
+          if (contracts.length > 0) {
+            schemaRegistryAddress = contracts[contracts.length - 1]?.address as Address;
+            break;
+          }
+        } catch (error) {
+          // Continue waiting
+        }
+      }
+
+      if (!schemaRegistryAddress) {
+        throw new Error("Schema Registry deployment timed out - could not get contract address");
+      }
+
+      // Deploy EAS contract with correct Schema Registry address
+      const easResult = await this.portalClient.request(GraphQLOperations.mutations.deployEAS, {
+        from: deployerAddress,
+        constructorArguments: {
+          registry: schemaRegistryAddress,
+          forwarder: defaultForwarder,
+        },
+        gasLimit: defaultGasLimit,
+      });
+
+      const easResponse = easResult as {
+        DeployContractEAS?: { transactionHash?: string };
+      };
+
+      if (!easResponse.DeployContractEAS?.transactionHash) {
+        throw new Error("EAS deployment failed - no transaction hash returned");
+      }
+
+      const easTxHash = easResponse.DeployContractEAS.transactionHash;
+
+      // Wait for EAS deployment and get contract address
+      let easAddress: Address | null = null;
+      attempts = 0;
+
+      while (!easAddress && attempts < maxAttempts) {
```

Comment:
Instead of polling make use of the 'waitForTransactionReceipt' util

 * import { waitForTransactionReceipt } from "@settlemint/sdk-portal";
 *
 * const transaction = await waitForTransactionReceipt("0x123...", {
 *   portalGraphqlEndpoint: "https://example.settlemint.com/graphql",
 *   accessToken: "your-access-token",
 *   timeout: 30000 // 30 seconds timeout
 * });